zebrad/src/components/mempool/storage/eviction_list.rs

```diff
@@ -0,0 +1,107 @@
+//! [`EvictionList`] represents the transaction eviction list with
+//! efficient operations.
+use std::{
+    collections::{HashMap, VecDeque},
+    time::{Duration, Instant},
+};
+
+use zebra_chain::transaction;
+
+/// An eviction list that allows to efficiently add entries, get entries,
+/// and remove older entries in the order they were inserted.
+pub struct EvictionList {
+    // Maps each TXID In the list to the instant they were added.
+    unique_entries: HashMap<transaction::Hash, Instant>,
+    // The entries in the order they were inserted.
+    // This can be larger than `unique_entries` if a same txid is added
+    // multiple times. Its instant will be overwritten in
+    // `unique_entries` but all entries will kept in `unique_entries`.
+    ordered_entries: VecDeque<(transaction::Hash, Instant)>,
+    // The maximum size of `unique_entries`.
+    size: usize,
```

Comment:
Nitpick: We usually use `size` or `len` for the current size, and `max_size` for the limit.

```suggestion
    max_size: usize,
```